CSharp/BatchExplorer/ViewModels/MainViewModel.cs

```diff
@@ -733,6 +733,107 @@ public CommandBase RefreshJobs
         }
 
         /// <summary>
+        /// Delete selected jobs
+        /// </summary>
+        public CommandBase DeleteSelectedJobs
+        {
+            get
+            {
+                return new CommandBase( 
+                    async (commandArg) =>
+                    {                        
+                        if (this.jobs.Any(a => a.IsChecked))
+                        {
+                            IEnumerable<string> deletedIds = this.jobs.Where(a => a.IsChecked).Select(s=>s.Id);
+                            var result =
+                            MessageBox.Show(
+                                deletedIds.Count() > 1 ? 
+                                $"Are you sure you want to delete {deletedIds.Count()} selected jobs?" :
+                                "Are you sure you want to delete the job?",
+                                "Confirm Delete",
+                                MessageBoxButton.YesNo);
+
+                            if (result == MessageBoxResult.Yes)
+                            {
+                                try
+                                {
+                                    this.LeftSpinnerIsVisible = true;
+                                    FirePropertyChangedEvent("LeftSpinnerIsVisible");
+
+                                    foreach (var job in this.jobs)
+                                    {
+                                        if (job.IsChecked)
+                                        {
+                                            await job.DeleteAsync(false);
+                                        }
+                                    }
+
+                                    //This is very hacky as the tab count never updated correctly if we refresh

```

Comment:
As this is public code, and a sample which customers will mine, we should do a polish pass on comments like this, e.g. convert "This is very hacky" to "NOTE: There is an outstanding issue where the tab count..." (ideally raising a GitHub issue for it and linking that in the comment), and fix typoes such as "issueing" and "inspite".